test/e2e/unmanaged_dns_test.go

```diff
@@ -240,8 +263,35 @@ func TestUnmanagedDNSToManagedDNSInternalIngressController(t *testing.T) {
 		t.Fatalf("failed to update ingresscontroller %s: %v", name, err)
 	}
 
-	if err := kclient.Delete(context.TODO(), lbService); err != nil && !errors.IsNotFound(err) {
-		t.Fatalf("failed to delete svc %s: %v", lbService.Name, err)
+	var oldLoadBalancerStatus corev1.LoadBalancerStatus
+	lbService.Status.LoadBalancer.DeepCopyInto(&oldLoadBalancerStatus)
+
+	// Only delete the service on platforms that don't automatically update the service's scope.
+	switch platform {
+	case configv1.AlibabaCloudPlatformType, configv1.AWSPlatformType, configv1.IBMCloudPlatformType, configv1.PowerVSPlatformType:
+		if err := kclient.Delete(context.TODO(), lbService); err != nil && !errors.IsNotFound(err) {
+			t.Fatalf("failed to delete svc %s: %v", lbService.Name, err)
+		}
```

Comment:
There's already a wait after the polling loop.  That should be enough:

1. Verify the service reports a different LB endpoint and is now internal.  
2. Verify the operator doesn't report that it is still updating the configuration.  
3. Verify the dnsrecord now reports "Managed".  
4. Verify we can connect to the LB at its updated address.  

Is there a need to repeat one of those checks, or is there some missing check?

Reply:
Oh sorry I commented over here https://github.com/openshift/cluster-ingress-operator/pull/845/files#r999787438, I think the comments said the same thing.

Reply:
If it doesn't matter whether the service was actually deleted or not, and it has no side effects, then you don't need to wait.

Reply:
Oh sorry, this comment was different, ignore my comment above.

We do wait on [Line 278](https://github.com/openshift/cluster-ingress-operator/pull/845/files#diff-a660dcb9b4adcc19e854b3f294ec260efabed12bb55eb2f2b8fb03ff6eb92527R278) to make sure the service was updated. We don't need to specifically wait for the service to be deleted, since we just want to trigger an update, and on some platform that's how you need to do it.

Reply:
It still isn't clear to me whether there is a missing wait, or an extra wait.  If there is a missing wait, it isn't clear to me what it would be or why the test is passing now.  If there is an extra wait, I'm not too bothered by it since the test is passing now.  

Remember, the goal here is to get CI working again ASAP so that we can be confident that we haven't broken anything, and have time to fix anything that is broken before branching day; barring a major problem with the test (such as breaking it for some job or changing it in a way such that it doesn't cover the intended functionality at all), I'd rather not delay it for issues that can be addressed in a future clean-up.

Reply:
As an interim, lgtm.